platforms/ide/problems-rendering/src/main/java/org/gradle/problems/internal/rendering/ProblemRenderer.java

```diff
@@ -41,28 +37,19 @@ public ProblemRenderer(Writer writer) {
     }
 
     public void render(List<Problem> problems) {
-        Map<ProblemId, List<Problem>> renderingGroups = new HashMap<>();
-        for (Problem problem : problems) {
-            List<Problem> groupedProblems = renderingGroups.computeIfAbsent(
-                problem.getDefinition().getId(),
-                id -> new ArrayList<>()
-            );
-            groupedProblems.add(problem);
-        }
-
-        renderingGroups.forEach((id, groupedProblems) -> renderProblemGroup(output, id, groupedProblems));
+        render(output, problems);
     }
 
     public void render(Problem problem) {
-        this.render(Collections.singletonList(problem));
+        render(Collections.singletonList(problem));
     }
 
-    static void renderProblemGroup(PrintWriter output, ProblemId id, List<Problem> groupedProblems) {
+    private static void render(PrintWriter output, List<Problem> problems) {
         String sep = "";
-        for (Problem problem : groupedProblems) {
+        for (Problem problem : problems) {
             output.printf(sep);
             renderProblem(output, problem);
-            sep = "%n";
```

Comment:
ℹ️ Just if you don't know, the `printf` with an `%n` will result in the `System.lineSeparator()`.
See: https://docs.oracle.com/javase/8/docs/api/java/util/Formatter.html

Reply:
Right, makes sense. I'll restore the previous state.

Reply:
As you wish, I really have no opinion here as both solutions reach the same result.
Do what you are more comfortable with.

Reply:
Your approach is perfectly fine. I just had some back and forth with windows test failures on line endings. I figured that `StringGroovyMethods.normalize()` is the thing I'm looking for and I forgot to revert the newline changes in this class.